### PR TITLE
Add address and email to PAYONE form

### PR DESCRIPTION
**Pull request by cmaier**
For various reasons _(see APPS-1296 in JIRA)_, we have to provide the credit card owners address and email during the addition of the card and in an encrypted form on usage. To do so a necessary form field and implementation processing this data has been added. There is no further validation of the input, except there has to be any. The state only has to be choosen if they are provided for the selected country.

### How to test?

* Install this branch with one of the apps making use of the sdk.
* Choose the testing environment from dev settings.
* Add a new credit cards where the provider is PAYONE.
  * Check that the country the device is set to _(device's language settings)_ is the preselected one.
    * If none can be found the use has to choose one manually.
   * Choose a country where a state is mandatory _(e.g. US, IN, ID, JP, ...)_
   * Check that all field that are visible need to provide some data, otherwise an error toast will appear and card cannot be added.
   * Play around with the fields and selectors and try to break it.

### Definition of Done

- [x] Changelog gepflegt
- ~~Dokumentation gepflegt~~
- [x] Alle Anforderung des Issues sind erfüllt
- [x] Selbst greviewed _(aka [Self-Reviews (eng)](https://www.nerdwallet.com/blog/engineering/why-you-should-be-doing-self-reviews/))_
- [x] Review mit Product Owner _(Release-Variante o. Minified Build)_

#### App Tests
- [ ] Minified getestet? _(aka. Release Build)_
- [x] Environements beachtet _(Production/Staging)_
- [x] Unterstützte Sprachen sind getestet
- [x] Light-/Dark-Mode getestet
- [x] Edge-Cases getestet
- [x] Android API Levels wurden berücksichtigt _(minSdk?)_

### CODE REVIEW

File: ui/src/main/java/io/snabble/sdk/ui/payment/PayoneInputView.kt [hunk 1]
```diff
@@ -330,32 +354,38 @@ class PayoneInputView @JvmOverloads constructor(context: Context, attrs: Attribu
...
-                pseudocardpan = requireNotNull(pseudocardpan),
-                truncatedcardpan = requireNotNull(truncatedcardpan),
-                cardtype = requireNotNull(cardtype),
-                cardexpiredate = requireNotNull(cardexpiredate),
-                lastname = requireNotNull(lastname),
+                pseudoCardPan = requireNotNull(pseudoCardPan),
+                truncatedCardPan = requireNotNull(truncatedCardPan),
+                cardType = requireNotNull(cardType),
+                cardExpiryDate = requireNotNull(cardExpireDate),
+                name = requireNotNull(name),
```
Comment (Fabtron) [inline ui/src/main/java/io/snabble/sdk/ui/payment/PayoneInputView.kt:379]: If some of these values magically happens to be null the user is stuck in the ui without any feedback until he navigates back. Shouldn't we provide information that something went wrong or at least wrap it up with a try an catch block launching `fail()` with a reason?

**Review by Fabtron — CHANGES_REQUESTED** [review]
Works fine, just one small suggestion.

File: ui/src/main/java/io/snabble/sdk/ui/payment/PayoneInputView.kt (hunk 1 above)
Comment (cmaier) [reply ui/src/main/java/io/snabble/sdk/ui/payment/PayoneInputView.kt:379]: To be honest, it's been like this before. I acutally dislike using `requireNotNull()` at all. I just left it, as it is. Seems to work.

Comment (cmaier) [reply ui/src/main/java/io/snabble/sdk/ui/payment/PayoneInputView.kt:379]: > So just because it was like this before, we shouldn't change it? 🤔
Did I make this statement? What is your intention and goal?

> Its not working it's just doesn't crash 🙃
Can provide an implementation and how to test it? I didn't dive that deep into this Javascript implementation and Payone.

Comment (Fabtron) [reply ui/src/main/java/io/snabble/sdk/ui/payment/PayoneInputView.kt:379]: Yes,it was ur statement, to improve code as we can. I already suggested to wrap it in a try catch block so that we can inform the the user as done in `fail` or over the same way to make some toast appear. There is not much to dive in. 
If u don't see any additional value in these changes, u can merge this PR cause the mandatory features are working. 
Its up to u

**Review by Fabtron — APPROVED** [review]

File: ui/src/main/java/io/snabble/sdk/ui/payment/PayoneInputView.kt (hunk 1 above)
Comment (cmaier) [reply ui/src/main/java/io/snabble/sdk/ui/payment/PayoneInputView.kt]: I figured it out myself and tested it. In the case that Payone says the data is valid but something we need is
still missing, we will fail the process and inform the user. I've added a plain generic english only fail info.

Does this work for you?

Comment (Fabtron) [reply ui/src/main/java/io/snabble/sdk/ui/payment/PayoneInputView.kt]: Yes, that looks fine. Thank u !
